src/Faker/Provider/en_US/Person.php

```diff
@@ -116,4 +116,21 @@ public static function suffix()
     {
         return static::randomElement(static::$suffix);
     }
+
+    /**
+     * @example '444114444' or '444-11-4444'
+     * @param boolean|null $dashed true, false or null
+     */
+    public function ssn($dashed = false)
+    {
+        $ssn = static::randomNumber(9);
+        while (!preg_match('/(?!000)(?!666)(?!9)[0-9]{3}(?!00)[0-9]{2}(?!0000)[0-9]{4}/', $ssn)) {
```

Comment:
Howdy,

1/10 (the 9XX-XX-XXXXs) will fail and the extremely rare other cases. We run this routinely and it 90% runs once and 99% runs in two.

I'll look at pre-formatting.

Thanks for feedback!
oh4real